userspace/libsinsp/container_engine/containerd.h

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#pragma once
+
+class sinsp_container_info;
+class sinsp_threadinfo;
+
+#include <libsinsp/container_engine/containerd/containers.grpc.pb.h>
+#include <libsinsp/container_engine/containerd/images.grpc.pb.h>
+#include <libsinsp/container_engine/container_async_source.h>
+#include <libsinsp/container_engine/container_engine_base.h>
+#include <libsinsp/container_engine/sinsp_container_type.h>
+
+namespace ContainerdContainerService = containerd::services::containers::v1;
+namespace ContainerdImageService = containerd::services::images::v1;
+
+namespace libsinsp {
+namespace container_engine {
+
+struct containerd_lookup_request {
+	containerd_lookup_request(): container_type(CT_CONTAINERD), uid(0), request_rw_size(false) {}
+
+	containerd_lookup_request(const std::string& container_id_value,
+	                          sinsp_container_type container_type_value,
+	                          unsigned long uid_value,
+	                          bool rw_size_value):
+	        container_id(container_id_value),
+	        container_type(container_type_value),
+	        uid(uid_value),
+	        request_rw_size(rw_size_value) {}
+
+	bool operator<(const containerd_lookup_request& rhs) const {
+		if(container_id != rhs.container_id) {
+			return container_id < rhs.container_id;
+		}
+
+		if(container_type != rhs.container_type) {
+			return container_type < rhs.container_type;
+		}
+
+		if(uid != rhs.uid) {
+			return uid < rhs.uid;
+		}
+
+		return request_rw_size < rhs.request_rw_size;
+	}
+
+	bool operator==(const containerd_lookup_request& rhs) const {
+		return container_id == rhs.container_id && container_type == rhs.container_type &&
+		       uid == rhs.uid && request_rw_size == rhs.request_rw_size;
+	}
+
+	std::string container_id;
+	sinsp_container_type container_type;
+	unsigned long uid;
+	bool request_rw_size;
```

Comment:
This is actually unused; do you plan to add support for `update_with_size` API to containerd or we can drop it?